testing/sdk_tasks.py

```diff
@@ -119,6 +119,22 @@ def fn():
         raise RuntimeError('Failed to kill task with pattern "{}", exit status: {}'.format(pattern, exit_status))
 
 
-def task_exec(task_name, command, **kwargs):
-    return sdk_cmd.run_cli(
-        "task exec {} {}".format(task_name, command), **kwargs)
+def task_exec(task_name : str, command : str) -> tuple:
+    """
+    Invokes the given command on the task via spawning `dcos task exec` in a subprocess.
+    :param task_name: Name of task to run command on.
+    :param command: The command to execute.
+    :return: a tuple consisting of the task exec's return code, stdout, and stderr
+    """
+    exec_cmd = "dcos task exec {} {}".format(task_name, command)
+    result = subprocess.run([exec_cmd], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
What was the motivation for this change? Are we not now duplicating the code in `sdk_cmd.run_cli` here? It also seems unrelated to the changes required for HDFS, as I only found calls that check `stdout`.

Reply:
I tried using `sdk_cmd.run_cli`, however the command string that it takes as input is then manipulated via [shakedown's `run_dcos_command`](https://github.com/dcos/shakedown/blob/2cfb691cffc9a76e8e4ba629bbedd463fc755221/shakedown/dcos/command.py#L103) and even after wrestling with putting the right quotation marks and escapes in the right places (which made things ugly), the command was still failing. However, running the exact same command simply via the shell worked perfectly as expected. @nickbp ran into this issue as well recently.

Even though shakedown is essentially [performing a subprocess call as well](https://github.com/dcos/shakedown/blob/2cfb691cffc9a76e8e4ba629bbedd463fc755221/shakedown/dcos/command.py#L122) it seems its internal formatting is somehow messing things up.

As for getting stderr, that's just a good practice to pass it along upstream.

Reply:
Could you 'upstream' this into sdk_cmd?

Reply:
Yes, then we would remove the shakedown dependency there too.